[TRCL-24] adjust streak cam alignment tab when using external spectrograph

pieleric opened the pull request:
Many small fixes, related to the streakcam alignment tab when the streakcam is not the "standard" spectrograph, but on the "spectrograph-dedicated".

pieleric committed:
[fix] SPARCv2 streak alignment shouldn't show big slit-in if it doesn't affect the streak ccd

pieleric committed:
[fix] SPARC2 alignment streak should only show lens mag if it can be changed

If there is only one choice, no need to show it in the GUI.

pieleric committed:
[refactor] remove uncessary attributes from the sparc2_align_tab

Unused attributes related to the streak cam -> remove them.

pieleric committed:
[feature] add tooltip about the shutter

Clarify that the shutter "checked" means the shutter is closed == no light.

tepals commented on src/odemis/acq/align/autofocus.py:
```diff
@@ -713,19 +703,51 @@ def GetSpectrometerFocusingDetectors(focuser):
+def _findSparc2Focuser(align_mode: str) -> model.Actuator:
+    """
+    Find the correct focus actuator for the given alignment mode, based on the microscope model.
+    :param align_mode: see Sparc2AutoFocus
```
If I check `Sparc2AutoFocus` I see the following options for align_mode: `OPM mode, spec-focus or spec-fiber-focus, streak-focus`. However here you also check for a few other modes. Should this docstring be extended, or the docstring for `Sparc2AutoFocus`, or something else?

tepals replied:
Additionally the docstring for `Sparc2AutoFocus` does not explain what align_mode is, could you update that?

tepals commented on src/odemis/gui/cont/tabs/sparc2_align_tab.py:
```diff
@@ -1485,71 +1488,69 @@ def _onManualFocus(self, event):
+            if opath == "streak-focus":
+                self.panel.slider_focus.Enable(True)
+                self.panel.cmb_focus_gratings.Enable(True)
+                # Do no enable detector selection, as only the streak-ccd is available
```
typo: no -> not

pieleric committed:
[fix] SPARC2 FSLT focus modes using the correct components

The manual focus of the streak cam would always turn on the brightlight,
while it might need the brighlight-ext.
=> change SPARC2 focus functions to automatically pick the correct
brightlight. That means the manual focus doesn't need to pass the
brightlight explicitly anymore.

The manual focus of the streak cam would also move always many axes
related to the specotgraph, even if the streak cam is on a separate
spectrograph => introduce the streak-focus-ext mode.

Also, pressing the "manual focus" button could block the GUI,
while the spectrograph is moving to wavelength 0.
=> rely on the optical path manager to switch to wavelength 0.

When manual focus is active, and switching mode, the brightlight would
stay on. => explicitly turn off the brightlight.

Also fix the OPM test cases.
